Kernel/LPO.cpp

```diff
@@ -176,11 +176,11 @@ Ordering::Result LPO::alpha(TermList* sl, unsigned arity, Term *t) const
   ASS(t->shared());
 
   for (unsigned i = 0; i < arity; i++) {
-    switch (lpo(t, *(sl - i))) {
```

Comment:
Isn't this OK? The arguments are switched, I agree, but on closer inspection the logic below is also switched to fix it. I like the `LPO::lpo(Term *, TermList)` function for when we know that the LHS is a compound term as we can skip a check in some cases. Maybe just the `return lexMAE` fix below was enough to fix the bug?

Reply:
Suppose some i exists such that s[i] > t. The issue is that in this case `lpo(t, s[i])` may still return `INCOMPARABLE` in some cases instead of `LESS` whereas with `lpo(s[i], t)` we should always get `GREATER` (assuming `lpo` abides by its specification). Does that make sense?

Reply:
Ahah, OK, that makes sense! Looks like we were being too clever then.

Reply:
Btw, does it make something less efficient that we create a `TermList` in two more cases in your opinion? Apart from the obvious creating one more object and some additional checks? Since we use `lpo` once with `(TermList, Term*)` and once with `(Term*, TermList)` either we implement it twice or have this version, but I would gladly hear that there is a third option.

Reply:
No, creating a `TermList` from a `Term *` is basically free I expect. It should be the size of a machine register with automatic storage, and the constructor just copies the pointer into a union. The compiler should completely elide it. The extra checks are very slightly less efficient but I expect some amount of compiler help/branch prediction with that. Overall what we have is fine, and I'll take correctness over (possibly, very slightly) a performance win and some maintenance burden.